src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1275,21 +1282,14 @@ namespace QgsWfs
         QDomElement bbElem = doc.createElement( QStringLiteral( "gml:boundedBy" ) );
         if ( format == QgsWfsParameters::Format::GML3 )
         {
-          // For WFS 1.1 we honor requested CRS and axis order
-          // Axis is not inverted if srsName starts with EPSG
-          // It needs to be an EPSG urn, e.g. urn:ogc:def:crs:EPSG::4326
-          // This follows geoserver convention
-          // See: https://docs.geoserver.org/stable/en/user/services/wfs/axis_order.html
+          // If requested SRS (outputSrsName) is different from rect CRS (crs) we need to transform the envelope
           const QString requestSrsName = request.serverParameters().value( QStringLiteral( "SRSNAME" ) );
-          const QString srsName = !requestSrsName.isEmpty() ? requestSrsName : crs.authid();
-          const bool invertAxis { mWfsParameters.versionAsNumber() >= QgsProjectVersion( 1, 1, 0 ) &&
-                                  crs.hasAxisInverted() &&
-                                  ! srsName.startsWith( QLatin1String( "EPSG:" ) ) };
+          const QString outputSrsName = !requestSrsName.isEmpty() ? requestSrsName : crs.authid();
+          const QgsCoordinateReferenceSystem outputCrs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( outputSrsName );
```

Comment:
Should we not take care of query parameter also like you did above? Maybe factorize this logic into a method?

Reply:
I don't think so. The response is built in 2 times. How to build: `<gml:boundedBy>` and how to `<gml:featureMember>`. This part is used to get the crs of the `<gml:boundedBy>`. 

I tried to change it to have the same logic everywhere. This works but this breaks existing unit tests and I have no rationale to explain why this change would be needed. I haven't been unable to find any answer in the specifications. I feel like we should be conservative here and this change might break existing user expectations or workflows.